config.go

```diff
@@ -465,17 +580,18 @@ func configure(v *viper.Viper, flags *pflag.FlagSet) {
 	v.SetDefault("sink.kafkaConnect.clickhouse.username", "default")
 	v.SetDefault("sink.kafkaConnect.clickhouse.password", "")
 
-	// Dedupe Redis configuration
-	v.SetDefault("dedupe.redis", false)
-	v.SetDefault("dedupe.redis.address", "127.0.0.1:6379")
-	v.SetDefault("dedupe.redis.database", 0)
-	v.SetDefault("dedupe.redis.username", "")
-	v.SetDefault("dedupe.redis.password", "")
-	v.SetDefault("dedupe.redis.expiration", "24h")
-	v.SetDefault("dedupe.redis.useSentintel", false)
-	v.SetDefault("dedupe.redis.masterName", "")
+	v.SetDefault("dedupe.enabled", false)
+	v.SetDefault("dedupe.driver", "memory")
 
 	// Dedupe Memory configuration
-	v.SetDefault("dedupe.memory", false)
-	v.SetDefault("dedupe.memory.size", 128)
+	v.SetDefault("dedupe.config.size", 128)
+
+	// Dedupe Redis configuration
+	v.SetDefault("dedupe.config.address", "127.0.0.1:6379")
```

Comment:
Any reason why `dedupe.config` namespace instead of `dedupe.redis`?

Reply:
That is to use the same config structure regardless of the driver:

```yaml
dedupe:
  enabled: true
  driver: redis
  config:
    address: ...

dedupe:
  enabled: true
  driver: memory
  config:
    size: ...

# VS

dedupe:
  enabled: true
  driver: redis
  redis: # This feels redundant
    address: ...
```